src/components/donation/Steps/DonateMethods/Stripe/Form.tsx

```diff
@@ -0,0 +1,166 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useEffect, useState } from "react";
+import { FormProvider, useForm } from "react-hook-form";
+import { Link } from "react-router-dom";
+import { ObjectSchema, number, object } from "yup";
+import { FormValues, Props } from "./types";
+import { SchemaShape } from "schemas/types";
+import { useStripeCurrenciesQuery } from "services/apes";
+import { useErrorContext } from "contexts/ErrorContext";
+import { Currency } from "components/CurrencySelector";
+import ExtLink from "components/ExtLink";
+import LoadText from "components/LoadText";
+import QueryLoader from "components/QueryLoader";
+import Split from "components/Split";
+import { CheckField, Field } from "components/form";
+import { requiredString } from "schemas/string";
+import { GENERIC_ERROR_MESSAGE } from "constants/common";
+import { appRoutes } from "constants/routes";
+import { TERMS_OF_USE_DONOR } from "constants/urls";
+import AdvancedOptions from "../../../AdvancedOptions";
+import CurrencySelectorField from "./CurrencySelectorField";
+
+type FormProps = Props & {
+  defaultValues: Partial<FormValues>;
+  onSubmit: (formValues: FormValues) => void | Promise<void>;
+};
+
+export default function Form(props: FormProps) {
+  const [countryCode, setCountryCode] = useState<string>();
+
+  const queryState = useStripeCurrenciesQuery(
+    { country_code: countryCode || "" },
+    { skip: !countryCode }
+  );
+  const { handleError } = useErrorContext();
+
+  useEffect(() => {
+    fetch("https://ipapi.co/json/")
```

Comment:
if `countryCode` is dependence of `stripeCurrency`, should merge to one query like `aws.bankingApplications.bankingApplication` , so that we don't fetch on every render